packages/flutter_tools/lib/src/version.dart

```diff
@@ -42,18 +35,47 @@ class FlutterVersion {
 
     _frameworkRevision = _runGit('git log -n 1 --pretty=format:%H');
     _frameworkAge = _runGit('git log -n 1 --pretty=format:%ar');
+
+    final String version = _runGit('git describe --match v*.*.* --exclude *-* --first-parent --long --tags');
+    final RegExp versionPattern = new RegExp('^v([0-9]+)\.([0-9]+)\.([0-9]+\)-([0-9]+)-g([a-f0-9]+)\$');
```

Comment:
It's hard to read this and know which fields represent what pieces.  Can you encapsulate this in a class that takes the output of the git command in its constructor and exposes meaningfully-named fields for each of the pieces it extracts from the regexp?

Reply:
What about using https://pub.dartlang.org/packages/pub_semver instead? (used in `pub` AFAIK)

Reply:
@zoechi Instead of what?

Reply:
Parsing the version, pumping major/minor/patch/dev, comparing versions
Just wanted to bring it to your attention in case you were not aware of it

Reply:
The version that is being parsed here isn't a semver version, it's a git tag description long format.

As far as adding 1 is concerned, I don't think we need to use a library for that. :-) We're trying to minimize our dependencies, so a library would have to do something a lot more complicated than that to justify using it.

Reply:
@tvolkert done